src/py/flwr/client/message_handler/task_handler.py

```diff
@@ -131,3 +132,35 @@ def wrap_client_message_in_task_res(client_message: ClientMessage) -> TaskRes:
         workload_id="",
         task=Task(ancestry=[], legacy_client_message=client_message),
     )
+
+
+def configure_task_res(
+    task_res: TaskRes, ref_task_ins: TaskIns, producer: Node
+) -> TaskRes:
+    """Set the metadata of a TaskRes.
+
+    Fill `group_id` and `workload_id` in TaskRes
+    and `producer`, `consumer`, and `ancestry` in Task in TaskRes.
+
+    `producer` in Task in TaskRes will remain unchanged/unset.
+
+    Note that protobuf API `protobuf.message.MergeFrom(other_msg)`
+    does NOT always overwrite fields that are set in `other_msg`.
+    Please refer to:
+    https://googleapis.dev/python/protobuf/latest/google/protobuf/message.html
+    """
+    task_res = TaskRes(
+        task_id="",  # This will be generated by the server
+        group_id=ref_task_ins.group_id,
+        workload_id=ref_task_ins.workload_id,
+        task=task_res.task,
+    )
+    # pylint: disable-next=no-member
+    task_res.task.MergeFrom(
+        Task(
+            producer=producer,
```

Comment:
Can we get this value from `task_ins.consumer`, or do we need to provide it manually?

Reply:
I think we can actually get producer from `task_ins`. The producer arg here is bc, conventionally, we obtain `node`  from `node: Node = cast(Node, node_store[KEY_NODE])` and set it to be the `producer`. I am okay with both options.

Reply:
Got it, let's leave it as-is